File: 2001-number-of-pairs-of-interchangeable-rectangles/2001-number-of-pairs-of-interchangeable-rectangles.cpp
class Solution {
public:
    long long interchangeableRectangles(vector<vector<int>>& r) {
        long long res=0;
        map<double,long long> m;
        for(int i=0;i<r.size();i++)
            m[r[i][0]/double(r[i][1])]++;
        
        for(auto i:m)
            res+=(i.second-1)*(i.second)/2;
        return res;
    }
};

File: 2001-number-of-pairs-of-interchangeable-rectangles/2001-number-of-pairs-of-interchangeable-rectangles.js
var interchangeableRectangles = function(rectangles) {
    const map=new Map()
    for(e of rectangles)
        map.set(e[0]/e[1],map.get(e[0]/e[1])==undefined?0:map.get(e[0]/e[1])+1)
    let res=0
    for(e of map){
        res+=(e[1]+1)*(e[1])/2
        }
    return res
};